crypto/store/store_lib.c

```diff
@@ -120,35 +120,43 @@ OSSL_STORE_open_ex(const char *uri, OSSL_LIB_CTX *libctx, const char *propq,
                 loader_ctx = loader->open(loader, uri, ui_method, ui_data);
         }
 #endif
-        if (loader == NULL
-            && (fetched_loader =
-                OSSL_STORE_LOADER_fetch(schemes[i], libctx, propq)) != NULL) {
-            const OSSL_PROVIDER *provider =
-                OSSL_STORE_LOADER_provider(fetched_loader);
-            void *provctx = OSSL_PROVIDER_get0_provider_ctx(provider);
-
-            loader_ctx = fetched_loader->p_open(provctx, uri);
-            if (loader_ctx == NULL) {
-                OSSL_STORE_LOADER_free(fetched_loader);
-                fetched_loader = NULL;
-            } else if(!loader_set_params(fetched_loader, loader_ctx,
-                                         params, propq)) {
-                (void)fetched_loader->p_close(loader_ctx);
-                OSSL_STORE_LOADER_free(fetched_loader);
-                fetched_loader = NULL;
+        if (loader == NULL) {
+            if ((fetched_loader =
+                 OSSL_STORE_LOADER_fetch(schemes[i], libctx, propq)) != NULL) {
+                const OSSL_PROVIDER *provider =
+                    OSSL_STORE_LOADER_provider(fetched_loader);
+                void *provctx = OSSL_PROVIDER_get0_provider_ctx(provider);
+
+                loader_ctx = fetched_loader->p_open(provctx, uri);
+                if (loader_ctx == NULL) {
+                    OSSL_STORE_LOADER_free(fetched_loader);
+                    fetched_loader = NULL;
+                } else if(!loader_set_params(fetched_loader, loader_ctx,
+                                             params, propq)) {
+                    (void)fetched_loader->p_close(loader_ctx);
+                    OSSL_STORE_LOADER_free(fetched_loader);
+                    fetched_loader = NULL;
+                }
+                loader = fetched_loader;
+            } else {
+                ERR_raise_data(ERR_LIB_OSSL_STORE,
```

Comment:
If I understand the code correctly, in case there is no store loader at all it will add as many errors as there are schemes. I suppose this is just a cosmetic issue but...

Reply:
Note that `ossl_store_get0_loader_int()` will report an error for each iteration where it fails as well.
I believe that the sane solution is to surround the loop with a ERR_set_masrk() / ERR_pop_to_mark() pair

Reply:
Also...  I'm not quite sure why we have this extra error report at all.  `OSSL_STORE_LOADER_fetch` *does* generate an error already!  So it's possible that the correct solution to this problem is to modify it to be more friendly...